app/pb_kits/playbook/pb_distribution_bar/_distribution_bar.scss

```diff
@@ -0,0 +1,31 @@
+@import "../tokens/colors";
+
+[class^=pb_distribution_bar] {
```

Comment:
It took me awhile to work through this, but this is what I am understanding these rules to try to generate.

```
[class^=pb_distribution_bar] {
  // common rules for pb distribution bar

  &[class*=_sm] {
    // specific pb distribution bar small rules
  }
  &[class*=_lg] {
    // specific pb distribution bar large rules
  }
}
```
To avoid writing out

```
.pb_distribution_bar_sm {
   // specific pb distribution bar small rules
  // common rules for pb distribution bar
}

.pb_distribution_bar_lg {
  // specific pb distribution bar large rules
  // common rules for pb distribution bar
}
```

Have we considered a mixin to allow the two rule to share the common rules?

I find this code very difficult to read and understand as it is written. I also have concerns with the grep-ability of these kids of rules. Have we considered the performance of rules written like this, I would be suspicious of the rules performance.

Reply:
It depends on the kit, but this reduces repeating common styles over and over. This is like using a BEM/Atomic Design approach and combining with class chaining in order (in a protected way) to avoid repeating common things over and over and bloating our output CSS.

We have an example similar to what you are describing and this produces very tight classes but lots of repeated styles.

https://github.com/powerhome/playbook/blob/master/app/pb_kits/playbook/pb_layout/_layout.scss
https://github.com/powerhome/playbook/blob/master/app/pb_kits/playbook/pb_layout/_layout_mixin.scss

This is still in flux and will continue to adapt and change as we grow playbook. We may go back to class chaining but using very specific classes. I'll give you a look at the PR's as we develop other approaches! 👊